services/director-v2/src/simcore_service_director_v2/core/dynamic_services_settings/sidecar.py

```diff
@@ -50,6 +48,55 @@ def enforce_r_clone_requirement(cls, v: int, values) -> PositiveInt:
         return v
 
 
+class PlacementConstraintStr(ConstrainedStr):
+    strip_whitespace = True
+    regex = re.compile(
+        r"^(?!-)(?![.])(?!.*--)(?!.*[.][.])[a-zA-Z0-9.-]*(?<!-)(?<![.])(!=|==)[a-zA-Z0-9_. -]*$"
+    )
+
+
+class PlacementSettings(BaseCustomSettings):
+    # This is just a service placement constraint, see
+    # https://docs.docker.com/engine/swarm/services/#control-service-placement.
+    DIRECTOR_V2_SERVICES_CUSTOM_CONSTRAINTS: list[PlacementConstraintStr] = Field(
+        default_factory=list,
+        example='["node.labels.region==east", "one!=yes"]',
+    )
+
+    DIRECTOR_V2_PLACEMENT_CONSTRAINTS_REPLACEMENTS_FOR_GENERIC_RESOURCES: dict[
+        str,
+        PlacementConstraintStr,
+    ] | None = Field(
+        default=None,
+        description=(
+            "Use placement constraints in place of generic resources, for details "
+            "see https://github.com/ITISFoundation/osparc-simcore/issues/5250 "
+            "When `None` (default), uses generic resources"
+        ),
+        example='{"AIRAM": "node.labels.CUSTOM==CUSTOM_VALUE"}',
```

Comment:
I think capital letters are not very well accepted for these. therefore I would not use them in the examples.